weather_dl/download_pipeline/config.py

```diff
@@ -109,3 +111,46 @@ def optimize_selection_partition(selection: t.Dict) -> t.Dict:
         del selection_['year']
 
     return selection_
+
+
+def prepare_partitions(config: Config) -> t.Iterator[Config]:
```

Comment:
I'm curious: why did you move these functions to this file? It seems like `partition.py` is a good place for them.

Reply:
This was to avoid a cyclic dependency issue. In `partitions.py` we got:
```python
from .parsers import prepare_target_name
```

Reply:
I just reverted that commit and confirmed it'd cause an `ImportError`